File: 3208-alternating-groups-ii/main.go
func numberOfAlternatingGroups(colors []int, k int) int {
    n := len(colors)
    ans, l := 0, 0

    for r := 1; r < n + k - 1; r++ {
        
        if colors[r % n] == colors[(r - 1) % n] {
            l = r
        }

        if r - l + 1 > k {
            l += 1
        }

        if r - l + 1 == k {
            ans++
        }

    }

    return ans
}

File: 3208-alternating-groups-ii/main.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors, int k) {
        
        int n = colors.size();
        int ans = 0, l = 0;

        for (int r = 1; r < n + k - 1; r++) {
            if (colors[r % n] == colors[(r - 1) % n]) {
                l = r;
            }

            if (r - l + 1 > k) {
                l++;
            } 

            if (r - l + 1 == k) {
                ans++;
            }
        }

        return ans;
    }
};
